### ISSUE TITLE
Canvas

### ISSUE THREAD

**Issue by Hedgepig** [witheve/Eve#511]
DOM can be very slow and not appropriate for more graphically intensive games, Canvas would be a better to work with.

**Comment by LandonPowell** [comment]
I second this. DOM is a recipe for disaster when it comes to a ton of moving objects on the screen.

**Comment by ibdknox** [comment]
Definitely on our list!

**Comment by olafleur** [comment]
And concretely, what are we talking about?
Converting all of Canvas API as close as possible in Eve? Or just using Canvas, but a different API?

For example, what would something like a circle drawing in a canvas look like as Eve syntax ?
(something like this, for example : http://www.html5canvastutorials.com/tutorials/html5-canvas-circles/ )

Are we looking to do something like this : 
```
bind @browser
   [#canvas width: 578 height: 200 children : 
     [#arc centerX: 100 centerY: 100 ...]

etc...
```
Or something totally different?

**Comment by Hedgepig** [comment]
This will definitely need a lot of consideration. I'd be tempted to say a separate API would be better. Something well suited to the domain language. But probably needs to be broad and general, then libraries can be made for more specific circumstances, e.g. games, graph creation, etc.

**Comment by RubenSandwich** [comment]
I second a separate API. While the DOM is slow it has baseline accessibility already built in. Screenreader support and other accessibility features to me are a must for any of the software I develop. (Yes Canvas has accessibility API's but they are off by default and very hard to work with and you usually just end up mostly mimicking the closet DOM elements you feel like your canvas element is like accessibility wise.)

**Comment by cmontella** [comment]
This is partially implemented with a low level api in Eve v0.3. 

See https://github.com/witheve/eve-starter/blob/master/programs/canvas-demo.ts for a demonstration.

**Comment by joshuafcole** [comment]
If anyone is interested in exploring higher level APIs the canvas watcher should be flexible enough to build on top of. Let me know what you come up with!
